src/averageAges.js

```diff
@@ -55,7 +77,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f')
+    .filter(mother =>
+      people.some(person => person.mother === mother.name));
+
+  let children = people.filter(person =>
+    mothers.some(mother => person.mother === mother.name));
+
+  if (onlyWithSon) {
+    children = children.filter(child => child.sex === 'm');
+  }
```

Comment:
If the flag is present, then you will run 2 filters. You can do a check inside the filter callback